text/1000-migrations.rst

```diff
@@ -332,47 +349,78 @@ The returned JSON conforms to the following pseudo-schema::
 Example::
 
     {
+      "parent": "m1vrzjotjgjxhdratq7jz5vdxmhvg2yun2xobiddag4aqr3y4gavgq",
+
       "confirmed": [
         "CREATE TYPE User { CREATE PROPERTY name -> str }"
       ],
 
-      "proposed": [
-        {
-          "statements": [{
-            "text": "ALTER TYPE Address " +
-                    "ALTER PROPERTY number " +
-                    "SET TYPE int64 USING \(expr)",
-            "required-user-input": [{
-              "name": "expr",
-              "prompt": "Altering Address.number to type " +
-                        "int64 requires an explicit conversion expression",
-            }]
-          }],
-          "confidence": 0.6
-          "prompt": "Did you alter the Address.number property?"
-        }, {
-          "statements": [{
-            "text": "ALTER TYPE Address { " +
-                    "DROP PROPERTY number; " +
-                    "CREATE PROPERTY number -> int64; }"
-          }],
-          "confidence": 0.4
-        }
-      ]
+      "proposed": {
+        "statements": [{
+          "text": "ALTER TYPE Address " +
+                  "ALTER PROPERTY number " +
+                  "SET TYPE int64 USING \(expr)",
+          "required-user-input": [{
+            "name": "expr",
+            "prompt": "Altering Address.number to type " +
+                      "int64 requires an explicit conversion expression",
+          }]
+        }],
+        "confidence": 0.6,
+        "prompt": "Did you alter the Address.number property?",
+        "safe": false
+      }
     }
 
-Migration operation classification
-----------------------------------
+The algorithm for obtaining the "proposed" data is as follows:
 
-The migrations system classifies the migrations operations into two categories:
-safe and unsafe, based on whether the operation is automatically reversible
-without losing any prior data.  For example, all ``CREATE`` operations are
-considered safe by definition, but also alterations to schema that doesn't
-involve data mutation, such as annotations, indexes, etc.  All other operations
-are classified as unsafe.
+1. Compute schema "A" from the state immediately preceding the
+   ``START MIGRATION`` command plus all statements in the "confirmed" list.
+
+2. Compute schema "B" from the SDL declaration passed to ``START MIGRATION``.
+
+3. Calculate the similarity matrix for each pair of objects of the same class
+   in schema "A" and schema "B" (e.g. compare every object type in schema "A"
+   with schema "B").  Only top-level objects are considered (i.e. object types,
+   abstract link definitions etc). The rows of the matrix represent objects
+   from "A", columns -- objects from "B", and the cells contain
```

Comment:
"The rows of the matrix represent objects from "A", columns -- objects"

not sure I can parse this sentence

Reply:
```
+-------------------------+
|      | Foo | Bar | Baz  |
+-------------------------+
|Foo   | 1.0 | 0.2 | 0.1  |
+-------------------------+
|Bar   | 0.2 | 0.9 | 0.3  |
+-------------------------+
```